contents/blog/secrets-of-posthog-query-performance.md

```diff
@@ -0,0 +1,152 @@
+---
+date: "2022-03-16" # :TODO
+title: "Secrets of PostHog query performance"
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+featuredImage: ../images/blog/launch-week-teaser.jpeg # :TODO:
+featuredImageType: full
+categories: ["Engineering", "Product updates", "Launch week"]
+author: ["karl-aksel-puulmann"]
+---
+
+We want PostHog to become the first choice for product analytics at any scale. To do that, users should have a smooth experience exploring their product data - including not waiting minutes for queries to load.
+
+In this post, I’m going to break down the why’s and how’s of several major performance improvements we've made in the last few months.
+
+## How does querying work within PostHog?
+
+The free-form querying experience in PostHog allows you to ask questions about your Trends, Funnels, Retention, and Cohorts with complicated filtering to top it off.
+
+These queries are processed by [ClickHouse](https://clickhouse.com/), where event, user, and group data is stored in a raw format without any preaggregation. ClickHouse excels at analyzing billions of events in seconds even with complex filtering.
+
+However, as awesome as ClickHouse is, nothing is without sharp edges and trade-offs.
+
+## Speeding up property filtering (up to) 25x
```

Comment:
"(up to) 25x" doesn't match the last sentence, where you say the average went up 55%. Should we have more consistent title naming (here we don't refer to the p99, other places we refer to avg in the percentage in the title)